### PR TITLE
Add/modify the v00 of draft.

### PR DESCRIPTION
**Pull request by reza-github**
Add/modify the v00 of draft.

### CODE REVIEW

**Review by prmanna — APPROVED** [review]

**Review by philbedard — APPROVED** [review]

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md [hunk 1]
```diff
@@ -108,58 +108,96 @@ contributor:
...
 
 normative:
   OIF-CMIS:
     title: "OIF Implementation Agreement (IA) Common Management Interface Specification (CMIS))"
     author:
-      org: OIF Forum
+    org: OIF Forum
     date: 27 April 2022
     seriesinfo: OIF CMIS IA
     target: https://www.oiforum.com/wp-content/uploads/OIF-CMIS-05.2.pdf
```
Comment (philbedard) [inline draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:137]: Can be updated in next revision but CMIS 5.3 has been published.

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md [hunk 2]
```diff
@@ -236,19 +277,19 @@ The document is divided into the following sections:
...
+{{figure-details-packet-optical-device}} shows a host packet device from vendor X, which is connected to optical device, equipped with optical pluggables from vendor X and Y. This figure exposes the following internal and external interfaces:
 
-1. This interface provides the control of packet device packet and optical functions. It provides these functions which can be decoupled such that there is no overlap between the optical and packet control functions.
+1. This interface provides the control of host packet and optical functions. It provides these functions which can be decoupled such that there is no overlap between the optical and packet control functions.
 
 2. The optical controller has read/write access to the optical network
 
-3. The CMIS (content management interoperability services) communication interface between hosts (e.g., packet devices) and optical pluggable
+3. The CMIS (content management interoperability services) communication interface between host packet devices) and optical pluggables
```
Comment (philbedard) [inline draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:286]: Need to fix the CMIS expanded naming to "Common Management Interface Specification"

**Review by philbedard — APPROVED** [review]
I saw a couple of minor typos in the document which I made comments on, otherwise it looks good.

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md (hunk 1 above)
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:137]: Done

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md (hunk 2 above)
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:286]: Done

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md [hunk 3]
```diff
@@ -349,7 +390,7 @@ The media side of the coherent pluggable is further divided into two functional
 
 ## Media Logical Channels
 
-The characterises of the Media Logical Channels are:
+The characterizes of the Media Logical Channels are:
```
[the remark this answers is not in the document]
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:393]: Done

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md [hunk 4]
```diff
@@ -108,58 +108,96 @@ contributor:
...
 
-This draft introduces the concpet of "Coherent Pluggable Manifest", where a represents the capabilities of a type&version of pluggable, maintained in a public repository of all pluggables. This document also covers gap analysis of current IETF drafts and other SDOs on coherent pluggables attributes and provides the complete lifecycle of a coherent pluggable from operators' approval through viability assessment to deployment.
+This draft introduces the concept of "Coherent Pluggable Manifest", where it represents the capabilities of pluggable, maintained in a repository. This document also covers gap analysis of current IETF drafts and other SDOs on coherent Pluggable attributes and provides the complete lifecycle of a coherent pluggable from operators' approval through viability assessment to deployment.
+
+This document also covers an analysis of the gap between current IETF drafts and the corresponding works in other standards bodies and industry. The lifecycle of a coherent pluggable including operators' approval, viability assessment to deployment and deployment is also covered.
```
Comment (sergiobelotti) [inline draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:200]: nothing is changed and the repetition is still present.

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md [hunk 5]
```diff
@@ -236,19 +277,19 @@ The document is divided into the following sections:
...
 
 5. Optical fiber connecting the optical devices to optical pluggables. This carries the flow of photonic signal from the optical device to the coherent pluggables. {{building-blocks}} will discuss this in more details.
 
-This draft presents the modeling of the coherent pluggable such as #1 and #2 in {{figure-details-packet-optical-device}} within a host packet device. The model presented in {{data-model}} consolidates properties of coherent pluggable on interfaces (4) and (5) where interface (5) provides the photonic/optical attributes and interface provides the host/electrical attributes.
+This draft presents the modeling of the coherent pluggable such as (4) and (5) in {{figure-details-packet-optical-device}} within a host packet device. The model presented in {{data-model}} consolidates properties of coherent pluggable on interfaces (4) and (5) where interface (5) provides the photonic/optical attributes and interface provides the host/electrical attributes.
```
Comment (sergiobelotti) [inline draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:292]: Nothing changed , you need to add interface 4 !!!!

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md [hunk 6]
```diff
@@ -369,21 +410,23 @@ The host side of the coherent pluggable is further divided into two functional b
 
 The characteristics of the Electrical Channels are:
 
-* The host interface lanes of the device forming the physical interface to the host platform data path device(s)
+Note that purpose of this section is to clarify the coherent pluggable. This purpose of this draft is not to define the data model of the electrical channels.
```
Comment (sergiobelotti) [inline draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:413]: for me the wording is still not clear. Please follow my suggestion above.

**Review by sergiobelotti — CHANGES_REQUESTED** [review]

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md [hunk 7]
```diff
@@ -108,58 +108,96 @@ contributor:
...
+     title: "A YANG model to manage the optical interface parameters for an external transponder in a WDM network"
+     date: 2025-01-06 
+     target: https://datatracker.ietf.org/doc/draft-ietf-ccamp-dwdm-if-param-yang/
+
+  
 
 --- abstract
 
-This draft outlines the modeling of optical pluggables within a host packet device in the context of a packet over optical network. The model encompasses all pertinent properties of the pluggable for various packet over optical use cases and is partitioned into three primary areas: the optical media interface, the electrical plug to host interconnect, and the physical equipment of the pluggable.
+This draft outlines the modeling of optical pluggables within a host packet device in the context of a packet over optical network. The model encompasses all pertinent properties of the pluggable for various packet over optical use cases and is partitioned into three primary areas: the optical media side, the electrical plug to host interconnect, and the physical equipment of the pluggable.
```
[the remark this answers is not in the document]
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:192]: sure we can discuss this. 
IMO, Electrical side is not needed for configuration but is needed for telemetry PM and alarms (for troubleshooting)

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:200]: You are right. It is removed.

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md [hunk 8]
```diff
@@ -236,19 +277,19 @@ The document is divided into the following sections:
 
 # Optical Pluggable in a Device with Packet Functions
 
-{{figure-details-packet-optical-device}} shows a packet device from vendor X, which is connected to optical device, equipped with optical pluggables from vendor X and Y. This figure exposes the following internal and external interfaces:
+{{figure-details-packet-optical-device}} shows a host packet device from vendor X, which is connected to optical device, equipped with optical pluggables from vendor X and Y. This figure exposes the following internal and external interfaces:
 
-1. This interface provides the control of packet device packet and optical functions. It provides these functions which can be decoupled such that there is no overlap between the optical and packet control functions.
+1. This interface provides the control of host packet and optical functions. It provides these functions which can be decoupled such that there is no overlap between the optical and packet control functions.
```
[the remark this answers is not in the document]
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:282]: Yes it is possible.
This can be one deployment model (aka Realization) operator might use. There might be other deployments. The point of this sentence is just separation of responsibilities for IP and Optical

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:292]: Nice catch. Added

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:413]: Agreed. Replaced

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md [hunk 9]
```diff
@@ -397,177 +440,502 @@ The "Equipment functional block" in {{figure-optical-pluggable-building-blocks}}
...
 
-The supported-modes are read-only capability attributes that define the functional capabilities of coherent pluggables. In other words, the coherent pluggable functional capabilities are described by a set of supported-modes, which includes read-only attributes such as modulation, bit rate, baud rate, chromatic dispersion, polarization, and FEC. For some attributes it includes value range (min. max) as well. A coherent pluggable may support multiple supported-modes, where each mode can be defined by one of the following methods:
+From a coherent pluggable data modeling perspective, a set of attributes is grouped together and represented by a single identifier known as the "Supported Mode." In essence, each supported mode encapsulates a combination of functional capabilities for coherent pluggables, such as modulation type, bit rate, baud rate, chromatic dispersion, polarization, FEC, and more. Some of these attributes may also include value ranges (e.g., minimum and maximum). A coherent pluggable can support multiple supported modes, each of which can be defined by one of the following methods:
```
[the remark this answers is not in the document]
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:457]: Sergio, since this needs for discussion, my suggestion is to keep this comment open and have  more discussions during our weekly meetings.

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md [hunk 10]
```diff
@@ -397,177 +440,502 @@ The "Equipment functional block" in {{figure-optical-pluggable-building-blocks}}
...
 
-* STANDARD: Defined by a Standards Developing Organization (SDO) such as ITU-T
-* NON-STANDARD: Defined by a forum such as Optical Internetworking Forum (OIF), OpenConfig or by an individual vendor or by an operator
+* Method 1: Defined by a Standards Developing Organization (SDO) such as ITU-T
```
[the remark this answers is not in the document]
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:459]: This is the intent. 
So, would you suggest what is needed to be added/replaced here?

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md [hunk 11]
```diff
@@ -397,177 +440,502 @@ The "Equipment functional block" in {{figure-optical-pluggable-building-blocks}}
...
+* Method 2: Defined by a forum such as Optical Internetworking Forum (OIF), a group of operators (OpenConfig) or by an individual vendor
```
[the remark this answers is not in the document]
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:460]: As previous comment, let's have this item as one of the topics for discussion during our weekly meetings.

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md [hunk 12]
```diff
@@ -397,177 +440,502 @@ The "Equipment functional block" in {{figure-optical-pluggable-building-blocks}}
...
+These methods define a namespace for the authority that specified the particular supported mode, whether it be a standards body, an association, a group of operators, or an individual vendor. For example, the supported modes in method 1 are well-known capabilities established by standards bodies such as the ITU-T {{G.698.2}}. These modes are recognized and supported by coherent pluggables, routers, and SDN controllers. This draft also supports the work of other Standards Development Organizations (SDOs), forums, or individual vendors as their specifications become available. In contrast, the supported-modes in method 2 are capabilities specified by forums such as OIF {{OIF-400ZR}}, OpenConfig, or by individual vendors. These supported-modes might not be supported by all coherent-pluggables, routers or SDN controllers. In specific, the supported-modes defined by an individual vendor might be not be recognized and supported by any routers and SDN controllers.
```
[the remark this answers is not in the document]
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:462]: Similar to previous comment, let's have this item as one of the topics for discussion during our weekly meetings.

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md [hunk 13]
```diff
@@ -397,177 +440,502 @@ The "Equipment functional block" in {{figure-optical-pluggable-building-blocks}}
...
 
-In contrast, the NON-STANDARD defined supported-modes are capabilities specified by forums such as OIF {{OIF-400ZR}}, OpenConfig, or by individual vendors. These supported-modes might not be supported by all coherent-pluggables, routers or SDN controllers. In specific, the supported-modes defined by an individual vendor might be not be recognized and supported by any routers and SDN controllers.
+It is important to note that, from a functional perspective of coherent pluggables, the authority that defined the supported modes is less critical than the specific attributes associated with each supported mode. The key consideration is the list of attributes defined by these supported modes.
```
[the remark this answers is not in the document]
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:464]: the intention is this:
Regardless of which authority defined this mode, the list of "Capability Attributes" represented by this mode are important.

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md [hunk 14]
```diff
@@ -397,177 +440,502 @@ The "Equipment functional block" in {{figure-optical-pluggable-building-blocks}}
...
+As illustrated in {{figure-plug-supported-mode}}, this document utilizes a consistent data structure for defining supported modes, irrespective of the authority that specified them. It includes:
 
-* supported-mode-id: ID of the supported mode.
-* organization-id: The authority that defines the supported-mode, such as ITU-T, OIF, OpenConfig, or a vendor. This provides a name space for an authority who defines the supported-mode.
-* supported-mode-type: The type of supported-mode which is STANDARD or NON-STANDARD
-* multiple tags: Optional tags that can be used for various purposes, mainly for future extensibility. For example, these tags can map the supported-mode to CMIS Media-ID. Other potential uses of this tag are reserved for future developments.
+* supported-mode-id: Identification of the supported mode
+* organization-id: This provides a name space for an authority who defines the supported-mode, i.e., the authority that defines the supported-mode, such as ITU-T, OIF, OpenConfig, or a vendor
```
[the remark this answers is not in the document]
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:469]: Similar to previous comment, let's have this item as one of the topics for discussion during our weekly meetings.

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md [hunk 15]
```diff
@@ -397,177 +440,502 @@ The "Equipment functional block" in {{figure-optical-pluggable-building-blocks}}
...
+* tags: Optional multiple tags that can be used for various purposes, mainly for future extensibility. For example, these tags can map the supported-mode to CMIS Media-ID (if they are not the same). Other potential uses of this tag are reserved for future developments.
```
[the remark this answers is not in the document]
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:470]: Are you referring to "tag"?
If so, the intention is to have something similar to service-tag introduced in https://datatracker.ietf.org/doc/html/draft-ietf-teas-ietf-network-slice-nbi-yang-07
Its purpose is to provide flexibility to the pluggable YANG for future extensibility. As described, one example might be mapping of supported-mode to CMIS media-ID.

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md (hunk 5 above)
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:292]: Can you suggest what is needed to be replaced/removed? I will revise the text

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md (hunk 6 above)
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:413]: Can you suggest an alternative text. I will add it

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md (hunk 4 above)
Comment (sergiobelotti) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:200]: It seems to me that this part "This document also covers an analysis of the gap between current IETF drafts and the corresponding works in other standards bodies and industry. The lifecycle of a coherent pluggable including operators' approval, viability assessment to deployment and deployment is also covered." is already reported before and it seems a repetition with other words. Am I missed something?

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md (hunk 5 above)
Comment (sergiobelotti) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:292]: "The model presented in {{data-model}} consolidates properties of coherent pluggable on interfaces (4) and (5) where interface (5) provides the photonic/optical attributes and interface provides the host/electrical attributes."
You need to add "....and interface (4) provides the host/electrical attributes"

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md (hunk 6 above)
Comment (sergiobelotti) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:413]: I would reword a bit : " Note that the purpose of this section is to clarify the role of electrical channel in the coherent pluggable"
and "The purpose of this draft is not ...."

Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:413]: Done

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md (hunk 4 above)
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:200]: this sentence outlines two aspects:
1- this talk about life cycle of coherent Pluggables
2- Covering Gap analysis
Both are not discussed in previous sentences.

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md (hunk 5 above)
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:292]: I have replaced the original text with your suggested one.

**Comment by ggalimba56** [comment]
Hi All,

I can’t attend the call today dua to a conflict.

Sorry about that,

Best Regards,

Gabriele Galimberti

Via Melgacciata, 13
20813 – Bovisio Masciago (MB)
Italy

+39 3357481947

Da: reza-github ***@***.***>
Risposta: italobusi/actn-wdm-pluggable-modelling ***@***.***>
Data: lunedì 21 ottobre 2024 alle ore 13:21
A: italobusi/actn-wdm-pluggable-modelling ***@***.***>
Cc: Gabriele Galimberti ***@***.***>, Review requested ***@***.***>
Oggetto: Re: [italobusi/actn-wdm-pluggable-modelling] Add/modify the v00 of draft. (PR #9)


@reza-github commented on this pull request.

________________________________

In draft-rokui-ccamp-actn-wdm-pluggable-modelling.md<https://github.com/italobusi/actn-wdm-pluggable-modelling/pull/9#discussion_r1808596674>:

>

 4. The data flow between the coherent pluggable and the packet data function through this interface. This is electrical interface between coherent pluggable and the host. {{building-blocks}} will discuss this in more details.



 5. Optical fiber connecting the optical devices to optical pluggables. This carries the flow of photonic signal from the optical device to the coherent pluggables. {{building-blocks}} will discuss this in more details.



-This draft presents the modeling of the coherent pluggable such as #1 and #2 in {{figure-details-packet-optical-device}} within a host packet device. The model presented in {{data-model}} consolidates properties of coherent pluggable on interfaces (4) and (5) where interface (5) provides the photonic/optical attributes and interface provides the host/electrical attributes.

+This draft presents the modeling of the coherent pluggable such as (4) and (5) in {{figure-details-packet-optical-device}} within a host packet device. The model presented in {{data-model}} consolidates properties of coherent pluggable on interfaces (4) and (5) where interface (5) provides the photonic/optical attributes and interface provides the host/electrical attributes.

I have replaced the original text with your suggested one.

**Review by vharish02 — APPROVED** [review]

File: draft-rokui-ccamp-actn-wdm-pluggable-modelling.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (reza-github) [reply draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:192]:
```suggestion
This draft outlines the modeling of optical pluggables within a host packet device in the context of a packet over optical network. The model encompasses all pertinent properties of the pluggable for various packet over optical use cases and is partitioned into three primary areas: the optical media side, the electrical plug to host interconnect, and the physical equipment of the pluggable..
```
